### ISSUE TITLE
Error Creating azurerm_service_plan using App Service Environment V3 using Isolated SKU

### ISSUE THREAD

**Issue by sahitya12** [hashicorp/terraform-provider-azurerm#17758]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

">= 1.0"

### AzureRM Provider Version

">=3.0.0"

### Affected Resource(s)/Data Source(s)

azurerm_service_plan

### Terraform Configuration Files

```hcl
Able to create with service_plan using other SKU's other than Isolated SKU I1v2, I2v2, I3v2

#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

# CREATE AN AZURE APP SERVICE PLAN
# This Terraform template configures an Azure App Service Plan

#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

data "azurerm_resource_group" "asp-rg" {
  name = var.resource_group_name
}

data "azurerm_app_service_environment_v3" "ase" {
  count               = var.enable_appservice_env == true ? 1 : 0
  name                = var.appservice_env_name
  resource_group_name = var.appservice_env_rg
}

#---------------------------------------------------

# CREATE NEW APP SERVICE PLAN

#---------------------------------------------------

resource "azurerm_service_plan" "asp" {
  for_each                     = var.app_service_plans
  name                         = each.value["name"]
  resource_group_name          = data.azurerm_resource_group.asp-rg.name
  location                     = data.azurerm_resource_group.asp-rg.location
  os_type                      = coalesce(lookup(each.value, "os_type"), "Linux")
  sku_name                     = coalesce(lookup(each.value, "sku_name"), "S1")
  app_service_environment_id   = try(data.azurerm_app_service_environment_v3.ase.0.id, null)
  maximum_elastic_worker_count = lookup(each.value, "maximum_elastic_worker_count", null)
  worker_count                 = lookup(each.value, "worker_count", null)
  per_site_scaling_enabled     = lookup(each.value, "per_site_scaling_enabled", null)
  zone_balancing_enabled       = lookup(each.value, "zone_balancing_enabled", null)
  tags                         = each.value["tags"]
}
```

### Debug Output/Panic Output

```shell
╷

│ Error: creating Service Plan: (Serverfarm Name "plte-dev-as-azp-use2-01" / Resource Group "ssc-ccoe-platengg-sub-dev-01-rg-use2-as-02"): web.AppServicePlansClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>
│
│ with azurerm_service_plan.asp["plan1"],
│ on main.tf line 20, in resource "azurerm_service_plan" "asp":
│ 20: resource "azurerm_service_plan" "asp" {

│
│ creating Service Plan: (Serverfarm Name "xxxxxxxxxxxxxxxxxxxxxxxx" /
│ Resource Group "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"):
│ web.AppServicePlansClient#CreateOrUpdate: Failure sending request:
│ StatusCode=0 -- Original Error: autorest/azure: Service returned an error.
│ Status=<nil> <nil>

╵
```

### Expected Behaviour

to create service plan using ISolated SKU

### Actual Behaviour

╷

│ Error: creating Service Plan: (Serverfarm Name "plte-dev-as-azp-use2-01" / Resource Group "ssc-ccoe-platengg-sub-dev-01-rg-use2-as-02"): web.AppServicePlansClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>
│
│ with azurerm_service_plan.asp["plan1"],
│ on main.tf line 20, in resource "azurerm_service_plan" "asp":
│ 20: resource "azurerm_service_plan" "asp" {

│
│ creating Service Plan: (Serverfarm Name "xxxxxxxxxxxxxxxxxxxxxxxx" /
│ Resource Group "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"):
│ web.AppServicePlansClient#CreateOrUpdate: Failure sending request:
│ StatusCode=0 -- Original Error: autorest/azure: Service returned an error.
│ Status=<nil> <nil>

╵

### Steps to Reproduce

terraform apply

### Important Factoids

EastUS2

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/service_plan

**Comment by shballa-msft** [comment]
Same issue i am also facing

**Comment by tombuildsstuff** [comment]
hey @sahitya12 

Taking a look through here this is an issue with the Azure SDK's Base Layer not correctly surfacing the error being returned from the API here - work to fix this is being tracked in #10858, as rather than having multiple issues tracking the same thing I'm going to close this issue in favour of #10858 - would you mind subscribing to that issue for updates?

Thanks!
